tests/unit/data_apis/test_create.py

```diff
@@ -0,0 +1,354 @@
+
+    name = "data-api"
+    instance_username = "neo4j"
+    instance_password = "password"
+    type_definitions = """type Movie {
+        title: String! @unique
+    }
+    """
+
+    result = runner.invoke(
+        create_data_api,
+        [
+            "--instance-id",
+            "123",
+            "--instance-username",
+            instance_username,
+            "--instance-password",
+            instance_password,
+            "--name",
+            name,
+            "--type-definitions",
+            type_definitions,
+        ],
+        obj=mock_data_api_config,
+    )
+
+    assert result.exit_code == 0
+    assert (
+        result.output
+        == f"""Could not open provided type definitions as file, trying as GraphQL SDL\n{printed_data(
+        {
+            "id": "321",
+            "name": "GraphQLDataConnector",
+            "type": "graphql",
+            "status": "creating",
+            "aura_instance": {"id": "123"},
+            "url": "https://321.b03bbb8d00271ccdcf4c289748fe43e2.graphql.neo4j-dev.io/graphql",
+        }
+    )}"""
+    )
+
+    api_request.assert_called_once_with(
+        "POST",
+        "https://graphql-api-staging.neo4j.io/v1/instances/123/data-apis",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer dummy-token",
+        },
+        timeout=10,
+        data=json.dumps(
+            {
+                "name": name,
+                "type": "graphql",
+                "aura_instance": {
+                    "username": instance_username,
+                    "password": instance_password,
+                },
+                "data_api": {
+                    "graphql": {
+                        "type_definitions": base64.b64encode(
+                            type_definitions.encode()
+                        ).decode()
+                    }
+                },
+            }
+        ),
+    )
+
+
+def test_create_data_api_with_invalid_type_definitions(
+    api_request, mock_data_api_config
+):
+    runner = CliRunner()
+
+    api_request.return_value = mock_response()
+
+    name = "data-api"
+    instance_username = "neo4j"
+    instance_password = "password"
+    type_definitions = """notatype Movie {
+        title: String! @unique
+    }
+    """
+
+    result = runner.invoke(
+        create_data_api,
+        [
+            "--instance-id",
+            "123",
+            "--instance-username",
+            instance_username,
+            "--instance-password",
+            instance_password,
+            "--name",
+            name,
+            "--type-definitions",
+            type_definitions,
+        ],
+        obj=mock_data_api_config,
+    )
+
+    print(result.output)
+
+    assert result.exit_code == 1
+    assert (
+        result.output
+        == """Could not open provided type definitions as file, trying as GraphQL SDL\nError: Could not parse as GraphQL\nError: An unexpected error occurred\n"""
```

Comment:
I feel like the "unexpected error" message could be improved and instead be clear that it is because the graphql is not valid

Reply:
As you can see in the middle, the correct error is displayed, for some reason though it still shows the "unexpected error" message. I don't believe this is anything I've added, fairly sure it was doing this the entire time.

Reply:
Yeah I see the correct error. Ideally we'd remove the unexpected error but don't worry if it takes too long to figure out.

It looks like the unexpected error occurred is the default error for the cli. Could it be because we are not catching an error when it is thrown?